apps/meteor/client/views/omnichannel/contactInfo/EditContactInfo.tsx

```diff
@@ -0,0 +1,282 @@
+	onCancel: () => void;
+};
+
+type ContactFormData = {
+	name: string;
+	emails: { address: string }[];
+	phones: { phoneNumber: string }[];
+	customFields: Record<any, any>;
+	contactManager: string;
+};
+
+const DEFAULT_VALUES = {
+	name: '',
+	emails: [],
+	phones: [],
+	contactManager: '',
+	customFields: {},
+};
+
+const getInitialValues = (data: ContactNewEditProps['contactData']): ContactFormData => {
+	if (!data) {
+		return DEFAULT_VALUES;
+	}
+
+	const { name, phones, emails, customFields, contactManager } = data ?? {};
+
+	return {
+		name: name ?? '',
+		emails: emails ?? [],
+		phones: phones ?? [],
+		customFields: customFields ?? {},
+		contactManager: contactManager ?? '',
+	};
+};
+
+const EditContactInfo = ({ contactData, onClose, onCancel }: ContactNewEditProps): ReactElement => {
+	const t = useTranslation();
+	const dispatchToastMessage = useToastMessageDispatch();
+	const queryClient = useQueryClient();
+	const handleNavigate = useContactRoute();
+
+	const canViewCustomFields = hasAtLeastOnePermission(['view-livechat-room-customfields', 'edit-livechat-room-customfields']);
+
+	const getContact = useEndpoint('GET', '/v1/omnichannel/contacts.get');
+	const createContact = useEndpoint('POST', '/v1/omnichannel/contacts');
+	const updateContact = useEndpoint('POST', '/v1/omnichannel/contacts.update');
+
+	const { data: customFieldsMetadata = [], isInitialLoading: isLoadingCustomFields } = useCustomFieldsMetadata({
+		scope: 'visitor',
+		enabled: canViewCustomFields,
+	});
+
+	const initialValue = getInitialValues(contactData);
+
+	const {
+		formState: { errors, isSubmitting },
+		control,
+		watch,
+		handleSubmit,
+	} = useForm<ContactFormData>({
+		mode: 'onBlur',
+		reValidateMode: 'onBlur',
+		defaultValues: initialValue,
+	});
+
+	const {
+		fields: emailFields,
+		append: appendEmail,
+		remove: removeEmail,
+	} = useFieldArray({
+		control,
+		name: 'emails',
+	});
+
+	const {
+		fields: phoneFields,
+		append: appendPhone,
+		remove: removePhone,
+	} = useFieldArray({
+		control,
+		name: 'phones',
+	});
+
+	const { emails, phones } = watch();
+
+	const validateEmailFormat = async (emailValue: string) => {
+		const currentEmails = emails.map(({ address }) => address);
+		const isDuplicated = currentEmails.filter((email) => email === emailValue).length > 1;
+
+		if (!validateEmail(emailValue)) {
+			return t('error-invalid-email-address');
+		}
+
+		try {
+			const { contact } = await getContact({ email: emailValue });
+			return (!contact || contact._id === contactData?._id) && !isDuplicated ? true : t('Email_already_exists');
+		} catch (error) {
+			return !isDuplicated ? true : t('Email_already_exists');
+		}
+	};
+
+	const validatePhone = async (phoneValue: string) => {
+		const currentPhones = phones.map(({ phoneNumber }) => phoneNumber);
+		const isDuplicated = currentPhones.filter((phone) => phone === phoneValue).length > 1;
+
+		try {
+			const { contact } = await getContact({ phone: phoneValue });
```

Comment:
This seems to return an error when trying to save a contact that does not have a phone set yet
![Screenshot 2024-10-18 at 15 09 55](https://github.com/user-attachments/assets/c453b6d7-be17-4814-86b3-ce7095922bb6)

Reply:
Resolved by https://github.com/RocketChat/Rocket.Chat/pull/32727/commits/bb2a324a32c429d944aad2c6ea8b8ca8a311ce64